src/_test_/Navbar.test.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { BrowserRouter } from 'react-router-dom';
+import renderer from 'react-test-renderer';
+import NavBar from '../components/navbar';
+
+describe('Navbar', () => {
+  test('should render correctly', () => {
+    const tree = renderer.create(
+      <BrowserRouter>
+        <NavBar />
+      </BrowserRouter>,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```

Comment:
- [ ] Kindly add some  unit tests for several components of the app and some pure functions in Redux code, as this is part of the core requirements